### PR TITLE
US9725 notification for form submission

### PR DESCRIPTION
**Pull request by vchtelmakh**
The purpose of this pull request is to create an EmailSender service, which can be used across all applications created by the CRDS development team. A lot of this code was copied from the Communication API, all of the calls were modified to use the `RequestBuilder`. 

I am NOT removing the original code from the comm API and pulling this in, because we decided we do not have time to appropriately test that aspect of it. This choice limits our testing to emails from FRED. 

- [ ] Is this safe from causing assembly issues somehow?

### CODE REVIEW

File: Crossroads.Web.Common/src/Configuration/CrossroadsWebCommonConfig.cs [hunk 1]
```diff
@@ -41,12 +41,18 @@ public static void Register(IUnityContainer container)
...
-            container.RegisterType<IMinistryPlatformRestRequestBuilderFactory, MinistryPlatformRestRequestBuilderFactory>(new ContainerControlledLifetimeManager(),
-                                                                                                                          new InjectionConstructor(
-                                                                                                                              container.Resolve<HttpClient>(
-                                                                                                                                  "ministryPlatformHttpClient"),
-                                                                                                                              container.Resolve<HttpClient>(
-                                                                                                                                  "authenticationHttpClient")));
+
+            container.RegisterType<IMinistryPlatformRestRequestBuilderFactory, 
+                MinistryPlatformRestRequestBuilderFactory>(
```
Comment (silbermm) [inline Crossroads.Web.Common/src/Configuration/CrossroadsWebCommonConfig.cs]: is this correct? does the `MinistryPlatformRestRequestBuilderFactory` take an `ApiUserRepository` and `IConfigurationRepository` as constructor args?

File: Crossroads.Web.Common/src/Helpers/Converters.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Crossroads.Service.Communication.Converters
```
Comment (silbermm) [inline Crossroads.Web.Common/src/Helpers/Converters.cs]: Should we make this namespace `Crossroads.Web.Common....` 
Not crucial, just a suggestion

File: Crossroads.Web.Common/src/MinistryPlatform/MPEmailSender.cs [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    private readonly IConfigurationWrapper _configurationWrapper;
+
+    private int? _authorUserId;
+    private int? _emailSenderContactId;
+
+    /// <summary>
+    /// Construct a new MPEmailSender.
+    /// </summary>
+    /// <param name="apiUserRepository">The IApiUserRepository that will be used to acquire an authenticated API token in order to send emails</param>
+    /// <param name="configurationWrapper">The IConfigurationWrapper that will be used to lookup any configuration values needed to send emails</param>
```
Comment (silbermm) [inline Crossroads.Web.Common/src/MinistryPlatform/MPEmailSender.cs]: missing a param in the documentation `IMinistryPlatformRestRequestBuilderFactory`

File: Crossroads.Web.Common/src/Helpers/Converters.cs (hunk 2 above)
Comment (vchtelmakh) [reply Crossroads.Web.Common/src/Helpers/Converters.cs]: Absolutely should, lazy copy pasta.

File: Crossroads.Web.Common/src/MinistryPlatform/MPEmailSender.cs (hunk 3 above)
Comment (vchtelmakh) [reply Crossroads.Web.Common/src/MinistryPlatform/MPEmailSender.cs]: Added!

File: Crossroads.Web.Common/src/Configuration/CrossroadsWebCommonConfig.cs (hunk 1 above)
Comment (vchtelmakh) [reply Crossroads.Web.Common/src/Configuration/CrossroadsWebCommonConfig.cs]: No, I am surprised it was working somehow, I am going to triple check, tests should have broken.

**Comment by vchtelmakh** [comment]
I need to learn to name things.

**Pull request closed without merging**